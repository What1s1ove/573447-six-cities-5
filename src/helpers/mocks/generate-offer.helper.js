import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
