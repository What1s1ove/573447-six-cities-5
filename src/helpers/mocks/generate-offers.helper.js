import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
