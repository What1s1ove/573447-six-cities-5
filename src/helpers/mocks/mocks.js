export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
