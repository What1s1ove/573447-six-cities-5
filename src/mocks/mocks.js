export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
