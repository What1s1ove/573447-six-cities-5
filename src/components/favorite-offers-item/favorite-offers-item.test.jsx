import * as React from 'react';
import {render} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import {mockedOffer} from '~/mocks/mocks';
import FavoriteOffersItem from './favorite-offers-item';

describe(`FavoriteOffersItem component`, () => {
  it(`should render FavoriteOffersItem component`, () => {
    const component = render(
        <MemoryRouter>
          <FavoriteOffersItem offer={mockedOffer} onFavoriteToggle={jest.fn()} />
        </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
