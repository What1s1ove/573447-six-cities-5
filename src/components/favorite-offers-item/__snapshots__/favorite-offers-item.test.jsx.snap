// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteOffersItem component should render FavoriteOffersItem component 1`] = `
<article
  class="favorites__card place-card"
>
  <div
    class="favorites__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer/1"
    >
      <img
        alt="Place image"
        class="place-card__image"
        height="110"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
        width="150"
      />
    </a>
  </div>
  <div
    class="favorites__card-info place-card__info"
  >
    <div
      class="place-card__price-wrapper"
    >
      <div
        class="place-card__price"
      >
        <b
          class="place-card__price-value"
        >
          €241
        </b>
        <span
          class="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        class="place-card__bookmark-button button"
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      class="place-card__rating rating"
    >
      <div
        class="place-card__stars rating__stars"
      >
        <span
          style="width:80%"
        />
        <span
          class="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      class="place-card__name"
    >
      <a
        href="/offer/1"
      >
        The Joshua Tree House
      </a>
    </h2>
    <p
      class="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
